nucleus/transform.py

```diff
@@ -0,0 +1,275 @@
+import numpy as np
+import transforms3d as t3d
+from pyquaternion import Quaternion
+
+
+class Transform:
```

Comment:
seems like the user can supply a lot of different values in the constructor. Do all class methods apply to all possible types of values?  if not, you may want to separate into different class types (maybe use inheritance?)